extensions/resource-deployment/src/interfaces.ts

```diff
@@ -93,12 +97,16 @@ export function instanceOfCommandDeploymentProvider(obj: any): obj is CommandDep
 	return obj && 'command' in obj;
 }
 
+export function instanceOfAzureSQLVMDeploymentProvider(obj: any): obj is AzureSQLVMDeploymentProvider {
+	return obj && 'azureSQLVMWizard' in obj;
+}
+
 export interface DeploymentProviderBase {
 	requiredTools: ToolRequirementInfo[];
 	when: string;
 }
 
-export type DeploymentProvider = DialogDeploymentProvider | BdcWizardDeploymentProvider | NotebookWizardDeploymentProvider | NotebookDeploymentProvider | WebPageDeploymentProvider | DownloadDeploymentProvider | CommandDeploymentProvider;
+export type DeploymentProvider = DialogDeploymentProvider | BdcWizardDeploymentProvider | NotebookWizardDeploymentProvider | NotebookDeploymentProvider | WebPageDeploymentProvider | DownloadDeploymentProvider | CommandDeploymentProvider | AzureSQLVMDeploymentProvider;
```

Comment:
It would be really awesome if we could stop special-casing everything. What was the blocker here that made it so you couldn't use one of the existing generic deployment types?

Reply:
There is a dependency tree in almost every page of the wizard. The current generic deployments do not offer support for such cases. Making a new deployment type was a design decision that I had to take to keeping the development time in mind. I am open to making a lot of fields generic (reusing stuff and also adding to the new features) now that the requirements are clear to me. The work is definitely not done yet and I will be actively contributing to the extension throughout the semester.

Reply:
One of the first task is to make this stuff more generic before the next release of ADS